source/android/mobile/src/main/java/io/adaptivecards/adaptivecardssample/MainActivityAdaptiveCardsSample.java

```diff
@@ -457,13 +483,15 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             case FILE_SELECT_CARD:
                 if (resultCode == RESULT_OK)
                 {
-                    loadAdaptiveCard(data);
+                    String fileName = data.getData().getLastPathSegment();
```

Comment:
this line repeats, you could move it before the switch starts

Reply:
We wouldn't want it to occur for any other onActivityResult's, that may be added to the switch